src/SignalR/common/Protocols.Json/src/Protocol/JsonHubProtocol.cs

```diff
@@ -124,9 +124,12 @@ private HubMessage ParseMessage(ReadOnlySequence<byte> input, IInvocationBinder
                 var hasArguments = false;
                 object[] arguments = null;
                 string[] streamIds = null;
-                JsonDocument argumentsToken = null;
-                JsonDocument itemsToken = null;
-                JsonDocument resultToken = null;
+                bool hasArgumentsToken = false;
+                Utf8JsonReader argumentsToken = default;
+                bool hasItemsToken = false;
+                Utf8JsonReader itemsToken = default;
+                bool hasResultToken = false;
+                Utf8JsonReader resultToken = default;
```

Comment:
This is kinda unfortunate. @ahsonkhan - is this the best pattern for "I maybe have a reader"?

Reply:
Yes, given its a struct. We also do not have the equality operators/IsDefault API since there wasn't a motivating scenario which showcased that such an API would add value: https://github.com/dotnet/corefx/issues/37534

Reply:
Is it too late to consider this a motivating scenario?

Reply:
> Is it too late to consider this a motivating scenario?

If we are asking for an `IsDefault`, not really, especially since this is quite a trivial API change. The only concern is that the impact of such an API is quite low, so I consider this really low priority (we can always add it in the future). Unless creating multiple `Utf8JsonReader` structs (as default), up front, is affecting performance and having an `IsDefault` check would help, I wouldn't prioritize fixing it for 3.0. In this case (we have several readers), so it might actually affect perf noticeably. 

@BrennanConroy, do you have a way to test the perf delta assuming such a property exists?

Also, wouldn't the compiler complain if you did this:
```C#
Utf8JsonReader resultToken;
if (resultToken.IsDefault) // Using unassigned variable.
{
 // ...
}
```

So, the question is what does the use case/scenario look like here? Do you have a proposal in mind? Looking more closely, it doesn't look like `IsDefault` would work for this case anyway?